python/pyspark/sql/dataframe.py

```diff
@@ -437,10 +437,11 @@ def hint(self, name, *parameters):
         if not isinstance(name, str):
             raise TypeError("name should be provided as str, got {0}".format(type(name)))
 
+        allowed_types = (basestring, list, float, int)
```

Comment:
Should it be `basestring` or `str` as previously here?

Reply:
It looks like Scala can handle unicode, so basestring looks correct. What you guys think?

Reply:
I think the point is that `basestring` is in python2 https://docs.python.org/2/library/functions.html#basestring

but not in python 3 https://docs.python.org/3.0/whatsnew/3.0.html#text-vs-data-instead-of-unicode-vs-8-bit

we should test this....

Reply:
At the start of the file, basestring, unicode and str becomes the same thing, so I don't think that the Python version can cause trouble at all.

Should I make tests for both types then? Any suggestion of how to test it?

Reply:
yap ok - https://github.com/DylanGuedes/spark/blob/f5de3b8fce22a2d0c55b86066471579c5720ec97/python/pyspark/sql/dataframe.py#L22

no, this should be good